controllers/lock_client.go

```diff
@@ -227,12 +188,48 @@ func (client *MockLockClient) Disabled() bool {
 	return !client.cluster.ShouldUseLocks()
 }
 
-// Close cleans up any resources that the client needs to keep open.
-func (client *MockLockClient) Close() error {
-	return nil
-}
-
 // NewMockLockClient creates a mock lock client.
 func NewMockLockClient(cluster *fdbtypes.FoundationDBCluster) (LockClient, error) {
 	return &MockLockClient{cluster: cluster}, nil
 }
+
+// fdbDatabaseCache provides a
+var fdbDatabaseCache = map[string]fdb.Database{}
+
+// getFDBDatabase opens an FDB database. The result will be cached for
+// subsequent calls, based on the cluster namespace and name.
+func getFDBDatabase(cluster *fdbtypes.FoundationDBCluster) (fdb.Database, error) {
+	cacheKey := fmt.Sprintf("%s/%s", cluster.ObjectMeta.Namespace, cluster.ObjectMeta.Name)
+	database, present := fdbDatabaseCache[cacheKey]
+	if present {
+		return database, nil
+	}
+
+	clusterFile, err := ioutil.TempFile("", "")
+	clusterFilePath := clusterFile.Name()
+	if err != nil {
+		return fdb.Database{}, err
+	}
+
+	defer clusterFile.Close()
+	if err != nil {
+		return fdb.Database{}, err
+	}
+
+	_, err = clusterFile.WriteString(cluster.Status.ConnectionString)
+	if err != nil {
+		return fdb.Database{}, err
+	}
+	err = clusterFile.Close()
```

Comment:
We close this file multiple times with the `defer` and in this statement. This shouldn't be a problem, we just should keep this in mind.